### PR TITLE
Implemented task 12

### PR DESCRIPTION
**Pull request by Heslav**

### CODE REVIEW

File: src/main/java/pl/programming4you/lectures5/lecture1/Task3.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 public class Task3 {
-    public static void main(String[] args) {
-        int firstNumber = 30;
-        int secondNumber = 15;
 
-        System.out.println(firstNumber + " plus " + secondNumber + " is " + (firstNumber + secondNumber));
-        System.out.println(firstNumber + " minus " + secondNumber + " is " + (firstNumber - secondNumber));
-        System.out.println(firstNumber + " times " + secondNumber + " is " + (firstNumber * secondNumber));
-        System.out.println(firstNumber + " divided by " + secondNumber + " is " + (firstNumber / secondNumber));
+    public String addition(int firstNumber, int secondNumber) {
```
Comment (programming4youpl) [inline src/main/java/pl/programming4you/lectures5/lecture1/Task3.java]: since this doesn't use any state of the calss Task3, you can (and probably should) make these methods static

File: src/test/java/pl/programming4you/lectures5/lecture1/Task4Test.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package pl.programming4you.lectures5.lecture1;
+
+import org.junit.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class Task4Test {
```
Comment (programming4youpl) [inline src/test/java/pl/programming4you/lectures5/lecture1/Task4Test.java:9]: add tests for empty and null string

File: src/test/java/pl/programming4you/lectures5/lecture4/Task8Test.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.junit.Test;
+import org.junit.jupiter.api.DisplayName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class Task8Test {
```
Comment (programming4youpl) [inline src/test/java/pl/programming4you/lectures5/lecture4/Task8Test.java:14]: add tests for 
- when the array is longer than the list
- when the array and list is empty
- when they are null

**Review by programming4youpl — CHANGES_REQUESTED** [review]

File: src/test/java/pl/programming4you/lectures5/lecture1/Task4Test.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package pl.programming4you.lectures5.lecture1;
+
+
+import org.junit.jupiter.api.Test;
+import pl.programming4you.lectures5.lecture4.Task8;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class Task4Test {
```
Comment (programming4youpl) [inline src/test/java/pl/programming4you/lectures5/lecture1/Task4Test.java:9]: add a test for null string, not only empty

File: src/test/java/pl/programming4you/lectures5/lecture4/Task8Test.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    public void shouldThrowIllegalArgumentExceptionForNull() {
+        assertThrows(IllegalArgumentException.class, () -> Task8.haveSameElements(null, null));
+    }
+
+    @Test
+    @DisplayName("Should throw an exception if list or array is empty")
+    public void shouldThrowIllegalArgumentExceptionForEmpty() {
+        List<String> list = new ArrayList<>();
+        String[] array = {};
+        assertThrows(IllegalArgumentException.class, () -> Task8.haveSameElements(list, array));
```
Comment (programming4youpl) [inline src/test/java/pl/programming4you/lectures5/lecture4/Task8Test.java]: this should return true

**Review by programming4youpl — CHANGES_REQUESTED** [review]

File: src/test/java/pl/programming4you/lectures5/lecture1/Task4Test.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    @Test
+    public void shouldReturnMessageLong() {
+        String longString = "Let's go Brandon";
+        String expectedMessage = "Your string is quite long";
+        assertEquals(expectedMessage, Task4.checkStringLength(longString));
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentExceptionForEmptyString() {
+        String randomText = "";
```
Comment (programming4youpl) [inline src/test/java/pl/programming4you/lectures5/lecture1/Task4Test.java]: is this a random text? inline this variable

**Review by programming4youpl — APPROVED** [review]
